modules/programs/taskwarrior.nix

```diff
@@ -0,0 +1,106 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.taskwarrior;
+
+  includeTheme = location:
+    if (location == null) then ""
+    else if (isString location) then "include ${pkgs.taskwarrior}/share/doc/task/rc/${location}.theme"
+    else "include ${location}";
+
+  formatValue = value:
+    if (isBool value) then (if value then "true" else "false")
+    else if (isList value) then (concatMapStringsSep "," formatValue value)
+    else "${value}";
+
+  formatLine = key: value:
+    "${key}=${formatValue value}";
+
+  formatSet = key: values:
+    (concatStringsSep "\n"
+      (mapAttrsToList
+        (subKey: subValue: formatPair "${key}.${subKey}" subValue)
+        values));
+
+  formatPair = key: value:
+    if (isAttrs value) then (formatSet key value)
+    else (formatLine key value);
+
+in
+
+{
+  options = {
+    programs.taskwarrior = {
+      enable = mkEnableOption "Task Warrior";
+
+      config = mkOption {
+        type = types.attrs;
+        default = {};
+        example = {
+          confirmation = false;
+          "report.minimal.filter" = "status:pending";
+          "report.active.columns" = [ "id" "start" "entry.age" "priority" "project" "due" "description" ];
+          "report.active.labels"  = [ "ID" "Started" "Age" "Priority" "Project" "Due" "Description" ];
```

Comment:
If the option supports recursive attribute sets then these keys may just as well be without quotes, i.e.,
```nix
report.minimal.filter = "status:pending";
report.active.columns = [ "id" "start" "entry.age" "priority" "project" "due" "description" ];
report.active.labels  = [ "ID" "Started" "Age" "Priority" "Project" "Due" "Description" ];
```